crates/store/src/genesis.rs

```diff
@@ -54,7 +54,7 @@ impl GenesisState {
         let header = BlockHeader::new(
             self.version,
             Digest::default(),
-            GENESIS_BLOCK,
+            GENESIS_BLOCK.into(),
```

Comment:
Not for this PR: but we probably should have made `GENESIS_BLOCK` into `BlockNumber` in `miden-base`.

Reply:
It is there as `BlockNumber::GENESIS`

Reply:
Actually, I created https://github.com/0xPolygonMiden/miden-base/pull/1088 to get rid of `GENESIS_BLOCK` constant. So, let's use `BlockNumber::GENESIS` here.

Reply:
I had thought of removing this on `miden-base` and was planning on mentioning it. I had also missed `BlockNumber::GENESIS` originally when doing this PR. Updated.